File: HPWorkpath_v1.6.3/Samples/ExampleAPIServices/source/EmailSample/src/com/hp/workpath/sample/emailsample/fragments/AddMailFragment.java
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.emailsample.fragments;

import android.app.Activity;
import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.text.TextUtils;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

import com.hp.workpath.sample.emailsample.EmailDialog;
import com.hp.workpath.sample.emailsample.MainActivity;
import com.hp.workpath.sample.emailsample.R;
import com.hp.workpath.sample.emailsample.interfaces.IDialogFragmentListener;
import com.hp.workpath.sample.emailsample.model.EmailAddress;

import java.util.HashMap;

public class AddMailFragment extends DialogFragment {
    private static final String TAG = MainActivity.TAG;
    private EditText mMailNameEditText;
    private EditText mMailAddressEditText;

    private EmailDialog.Type mAddMailType;

    private IDialogFragmentListener mListener;

    @Override
    public void onAttach(Activity activity) {
        super.onAttach(activity);
        try {
            mListener = (IDialogFragmentListener) activity;
        } catch (ClassCastException e) {
            Toast.makeText(activity, activity.getClass().getSimpleName()
                    + " must implement IDialogFragmentListener", Toast.LENGTH_SHORT).show();
            Log.e(TAG, activity.toString() + " must implement IDialogFragmentListener");
        }
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Bundle args = getArguments();
        if (args != null) {
            mAddMailType = (EmailDialog.Type) args.getSerializable(EmailDialog.DIALOG_TYPE);
        }
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        LayoutInflater inflater = getActivity().getLayoutInflater();
        View view = inflater.inflate(R.layout.dialog_add_mail, null);
        findViewElements(view);
        String title = getResources().getString(R.string.add);
        switch (mAddMailType) {
            case ADD_TO:
                title += " (" + getResources().getString(R.string.to) + ")";
                break;
            case ADD_CC:
                title += " (" + getResources().getString(R.string.cc) + ")";
                break;
            case ADD_BCC:
                title += " (" + getResources().getString(R.string.bcc) + ")";
                break;
        }
        AlertDialog.Builder dialogBuilder =
                new AlertDialog.Builder(getActivity(), R.style.DialogTheme)
                        .setTitle(title)
                        .setView(view)
                        .setPositiveButton(android.R.string.ok, mOKListener)
                        .setNegativeButton(android.R.string.cancel, mCancelListener)
                        .setCancelable(false);
        return dialogBuilder.create();
    }

    @Override
    public void onResume() {
        super.onResume();
        if (mListener == null) {
            dismiss();
        }
    }

    private void findViewElements(View view) {
        mMailNameEditText = view.findViewById(R.id.mailNameEditText);
        mMailAddressEditText = view.findViewById(R.id.mailAddrEditText);
    }

    private DialogInterface.OnClickListener mOKListener = new DialogInterface.OnClickListener() {
        @Override
        public void onClick(DialogInterface dialog, int which) {
            String mailName = mMailNameEditText.getText().toString();
            String mailAddress = mMailAddressEditText.getText().toString();
            if (!TextUtils.isEmpty(mailAddress)) {
                HashMap<String, Object> result = new HashMap<>();
                EmailAddress emailAddress = new EmailAddress(mailAddress, mailName);
                result.put(EmailDialog.DIALOG_TYPE, mAddMailType);
                result.put(getString(R.string.email), emailAddress);
                mListener.onReturnValue(result);
                getDialog().dismiss();
            } else {
                Toast.makeText(getActivity(), getString(R.string.email_address_empty),
                        Toast.LENGTH_LONG).show();
            }
        }
    };

    private DialogInterface.OnClickListener mCancelListener = new DialogInterface.OnClickListener() {
        @Override
        public void onClick(DialogInterface dialog, int which) {
            dialog.cancel();
        }
    };
}

File: HPWorkpath_v1.6.3/Samples_Kotlin/ExampleAPIServices/source/EmailSample/src/com/hp/workpath/sample/emailsample/fragments/AddMailFragment.kt
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.emailsample.fragments

import android.app.Activity
import android.app.Dialog
import android.content.Context
import android.content.DialogInterface
import android.os.Bundle
import android.text.TextUtils
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatDialogFragment
import com.hp.workpath.sample.emailsample.EmailDialog
import com.hp.workpath.sample.emailsample.MainActivity
import com.hp.workpath.sample.emailsample.R
import com.hp.workpath.sample.emailsample.databinding.DialogAddMailBinding
import com.hp.workpath.sample.emailsample.interfaces.IDialogFragmentListener
import com.hp.workpath.sample.emailsample.model.EmailAddress

class AddMailFragment : AppCompatDialogFragment() {

    private lateinit var mAddMailType: EmailDialog.Type
    private lateinit var mListener: IDialogFragmentListener
    private lateinit var customView: View
    private var mBindingFragment: DialogAddMailBinding? = null
    private val mBindingDialogAddMail get() = mBindingFragment!!

    override fun onAttach(context: Context) {
        super.onAttach(context)
        var activity: Activity? = null
        if (context is Activity) {
            activity = context
        }
        try {
            mListener = activity as IDialogFragmentListener
        } catch (e: ClassCastException) {
            Toast.makeText(activity, activity?.javaClass?.simpleName
                    + " must implement IDialogFragmentListener", Toast.LENGTH_SHORT).show()
            Log.e(TAG, "$activity must implement IDialogFragmentListener")
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val args = arguments
        if (args != null) {
            mAddMailType = args.getSerializable(EmailDialog.DIALOG_TYPE) as EmailDialog.Type
        }
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val inflater = requireActivity().layoutInflater
        mBindingFragment = DialogAddMailBinding.inflate(inflater)
        customView = mBindingDialogAddMail.root
        var title = resources.getString(R.string.add)
        when (mAddMailType) {
            EmailDialog.Type.ADD_TO -> title += " (" + resources.getString(R.string.to) + ")"
            EmailDialog.Type.ADD_CC -> title += " (" + resources.getString(R.string.cc) + ")"
            EmailDialog.Type.ADD_BCC -> title += " (" + resources.getString(R.string.bcc) + ")"
            else -> Unit
        }

        val dialogBuilder = AlertDialog.Builder(requireActivity(), R.style.DialogTheme)
                .setTitle(title)
                .setView(customView)
                .setPositiveButton(android.R.string.ok, mOKListener)
                .setNegativeButton(android.R.string.cancel, mCancelListener)
                .setCancelable(false)
        return dialogBuilder.create()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return customView
    }

    override fun onDestroyView() {
        super.onDestroyView()
        mBindingFragment = null
    }

    private val mOKListener = DialogInterface.OnClickListener { _, _ ->
        val mailName = mBindingDialogAddMail.mailNameEditText.text.toString()
        val mailAddress = mBindingDialogAddMail.mailAddrEditText.text.toString()
        if (!TextUtils.isEmpty(mailAddress)) {
            val result = HashMap<String, Any>()
            val emailAddress = EmailAddress(mailAddress, mailName)
            result[EmailDialog.DIALOG_TYPE] = mAddMailType
            result[requireActivity().getString(R.string.email)] = emailAddress
            mListener.onReturnValue(result)
            dialog?.dismiss()
        } else {
            Toast.makeText(activity, requireActivity().getString(R.string.email_address_empty),
                    Toast.LENGTH_LONG).show()
        }
    }
    private val mCancelListener = DialogInterface.OnClickListener { dialog, _ -> dialog.cancel() }

    companion object {
        private const val TAG = MainActivity.TAG
    }
}